pkg/gatherers/clusterconfig/gather_cluster_ingress_certificates.go

```diff
@@ -0,0 +1,236 @@
+				certInfo, certErr := getCertificateInfoFromSecret(ctx, coreClient, namespace, secretName)
+				if certErr != nil {
+					errs = append(errs, certErr)
+					continue
+				}
+
+				// Step 5: Add certificate info to the certificates list
+				c, exists := certificatesInfo[infereKey(certInfo)]
+				if exists {
+					c.Controllers = append(c.Controllers,
+						ControllerInfo{Name: controller.Name, Namespace: controller.Namespace},
+					)
+					//
+				} else {
+					certInfo.Controllers = append(certInfo.Controllers,
+						ControllerInfo{Name: controller.Name, Namespace: controller.Namespace},
+					)
+					certificatesInfo[infereKey(certInfo)] = certInfo
+				}
+			}
+		}
+	}
+
+	var ci []*CertificateInfo
+	if len(certificatesInfo) > 0 {
+		// Step 6: Generate the certificates record
+		ci = make([]*CertificateInfo, len(certificatesInfo))
+		i := 0
+		for _, v := range certificatesInfo {
+			ci[i] = v
+			i++
+		}
+	}
+
+	return ci, errs
+}
+
+func getCertificateInfoFromSecret(
+	ctx context.Context, coreClient corev1client.CoreV1Interface,
+	namespace, secretName string) (*CertificateInfo, error) {
+	//
+	secret, err := coreClient.Secrets(namespace).Get(ctx, secretName, metav1.GetOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to get secret '%s' in namespace '%s': %v", secretName, namespace, err)
+	}
+
+	crtData, found := secret.Data["tls.crt"]
+	if !found {
+		return nil, fmt.Errorf("'tls.crt' not found in secret '%s' in namespace '%s'", secretName, namespace)
+	}
+
+	block, _ := pem.Decode(crtData)
+	if block == nil {
+		return nil, fmt.Errorf("unable to decode certificate (x509) from secret '%s' in namespace '%s'", secretName, namespace)
+	}
+
+	cert, err := x509.ParseCertificate(block.Bytes)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse certificate from secret '%s' in namespace '%s': %v", secretName, namespace, err)
+	}
+
+	return &CertificateInfo{
+		Name:        secretName,
+		Namespace:   namespace,
+		NotBefore:   cert.NotBefore,
+		NotAfter:    cert.NotAfter,
+		Controllers: []ControllerInfo{},
+	}, nil
+}
+
+func getRouterCACertInfo(ctx context.Context, coreClient corev1client.CoreV1Interface) (*CertificateInfo, error) {
+	const (
+		routerCASecret    string = "router-ca"
+		routerCANamespace string = "openshift-ingress-operator"
+	)
+	certInfo, err := getCertificateInfoFromSecret(ctx, coreClient, routerCANamespace, routerCASecret)
+	if err != nil {
+		return nil, err
+	}
+
+	return certInfo, nil
+}
+
+func getRouterCertsDefaultCertInfo(ctx context.Context, coreClient corev1client.CoreV1Interface) (*CertificateInfo, error) {
+	const (
+		routerCertsSecret    string = "router-certs-default"
+		routerCertsNamespace string = "openshift-ingress"
+	)
+	certInfo, err := getCertificateInfoFromSecret(ctx, coreClient, routerCertsNamespace, routerCertsSecret)
+	if err != nil {
+		return nil, err
+	}
+
+	return certInfo, nil
+}
+
+func infereKey(info *CertificateInfo) CertificateInfoKey {
```

Comment:
nit: what about to call it something like `newCertificateInfoKey`?

Reply:
Solved in 82311f791c8e70eac909fe064f3a3eef081ea172